### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by canilo1**
Don't mind the other files; I was just deleting the CSS files so that I could see it. But now, you can't see it because the other files are blocking it. I'm too tired to figure out how to fix that, so to see it for now, just delete the other CSS files. But I did it! It looks almost identical to the Figma design. Any feedbaack would be nice.

### CODE REVIEW

**Comment by canilo1** [comment]
Heres a image of it 
image_2024-08-12_131152071
Let me know what yall think.

File: frontend/src/components/layout/SignUp.jsx
Comment (OGoncharenko) [inline frontend/src/components/layout/SignUp.jsx:10]: There is a typo in "fundamentals".
I believe that className should be in lower case. You have className={styles.header}, but className={styles.Headings}. The same with id.

File: frontend/src/components/layout/SignUp.jsx
Comment (ElijahMSmith) [inline frontend/src/components/layout/SignUp.jsx:6]: We don't need a fragment here since it only has one child element (the div).

File: frontend/src/components/layout/SignUp.jsx
Comment (ElijahMSmith) [inline frontend/src/components/layout/SignUp.jsx:10]: Extra \ here should be removed.

File: frontend/src/components/layout/SignUp.jsx
Comment (ElijahMSmith) [reply frontend/src/components/layout/SignUp.jsx:6]: Also, the indentation/formatting here is not quite right. Please make sure to install the `Prettier` extension and turn on `Autosave` and `Format on Save` in your settings.

File: frontend/src/components/layout/SignUp.jsx
Comment (ElijahMSmith) [reply frontend/src/components/layout/SignUp.jsx:10]: As a convention, we should make all our class names camelCase (starting with a lowercase).

File: frontend/src/pages/frontPage/Panel.jsx
Comment (ElijahMSmith) [inline frontend/src/pages/frontPage/Panel.jsx:3]: Let's add a blank line below here.

File: frontend/src/pages/frontPage/Panel.jsx
Comment (ElijahMSmith) [inline frontend/src/pages/frontPage/Panel.jsx:8]: Let's also fix the formatting in this file.

File: frontend/src/components/layout/Signup.module.css
Comment (ElijahMSmith) [inline frontend/src/components/layout/Signup.module.css:81]: Formatting here too!

**Review by ElijahMSmith — CHANGES_REQUESTED** [review]
Can we make this pop up in the center of the screen like in the Figma design before we merge this, ideally with a slightly blurred background as well? There should be some good resources online to help with centering the popup modal.

File: frontend/node_modules/.package-lock.json
Comment (ElijahMSmith) [inline frontend/node_modules/.package-lock.json:1]: I don't know how this file has gotten around the .gitignore, but it should not be committed. Please delete the node_modules folder, run `git add .` and commit the removal of this file, then run `npm i` again and hopefully it should not be tracked.

File: frontend/src/components/layout/SignUp.jsx
Comment (ElijahMSmith) [inline frontend/src/components/layout/SignUp.jsx:1]: This import does not work when I run `npm run dev` because the file is `Signup` when this imports `SignUp`. Let's name the file `SignUp.module.css` to be consistent with the component file name.

**Review by ElijahMSmith — APPROVED** [review]
Clear to merge once these comments are addressed. Make sure to squash your commits! I think I adjusted the settings on pull requests to give you the option by the merge button when you are ready to submit, or feel free to do it through your terminal.

File: frontend/src/pages/frontPage/Panel.jsx
Comment (ElijahMSmith) [inline frontend/src/pages/frontPage/Panel.jsx:10]: This is going to be really annoying to have over the screen all the time when it gets merged in. Can we find a way to show this only when someone clicks on the "register" nav link?

You could:
- Control whether this renders using a state variable like `showingSignupForm` that flips true/false when the nav link is clicked.
- Register the signup form at the `/register` route, which is currently when I am navigated when I click the register button (and that's an empty page right now since we don't have anything there).

I would lean towards the first since we've designed this to just be a popup in front of whatever else is on the screen. In that case, I would probably move this component into the `Nav` component where that state would live.

File: frontend/src/components/layout/SignUp.jsx
Comment (canilo1) [reply frontend/src/components/layout/SignUp.jsx:6]: Sure, i made almost everything lowecase, except the ones that have 2 names or so, for example handleVisibility(this is two words so its camel case) i hope thats what you mean

File: frontend/src/components/layout/Nav.jsx
Comment (canilo1) [reply frontend/src/components/layout/Nav.jsx:22]: This is what i did, 
const navLinks = isLoggedIn
    ? [
        { id: 1, link: "Dashboard" },
        { id: 2, link: "Account" },
        { id: 3, link: "Logout" }, ]
    : [
        { id: 1, link: "About" },
        { id: 2, link: "Login" },
        { id: 3, link: "Register" },];

**Pull request merged**
